modules/reactor/src/meshgenerators/PinMeshGenerator.C

```diff
@@ -139,165 +133,155 @@ PinMeshGenerator::PinMeshGenerator(const InputParameters & parameters)
   {
     mooseError("Region IDs must be assigned with parameter region_ids");
   }
+  if (isParamValid("block_names"))
+  {
+    _has_block_names = true;
+    _block_names = getParam<std::vector<std::vector<std::string>>>("block_names");
+    if (_region_ids.size() != _block_names.size())
+      mooseError("The size of block_names must match the size of region_ids");
+    for (const auto i : make_range(_region_ids.size()))
+      if (_region_ids[i].size() != _block_names[i].size())
+        mooseError("The size of block_names must match the size of region_ids");
```

Comment:
where is the check for size of region_ids being the same as the number of axial planes in extrusion

I think it's elswhere, down for a pointer

Reply:
Doesn't look like this check exists, will add this. Thanks for the catch

Reply:
Added a check for region_ids, background_regions_ids, and duct_region_ids